clang/lib/CIR/CodeGen/CIRGenStmtOpenMP.cpp

```diff
@@ -43,3 +91,34 @@ CIRGenFunction::buildOMPParallelDirective(const OMPParallelDirective &S) {
   builder.create<TerminatorOp>(getLoc(S.getSourceRange().getEnd()));
   return res;
 }
+
+mlir::LogicalResult
+CIRGenFunction::buildOMPTaskwaitDirective(const OMPTaskwaitDirective &S) {
+  mlir::LogicalResult res = mlir::success();
+  // Getting the source location information of AST node S scope
+  auto scopeLoc = getLoc(S.getSourceRange());
```

Comment:
Since the locations are only used for param passing, please delete the comment and just add the function call directly while passing the param. Same for the other 2 methods